packages/mendel-middleware/index.js

```diff
@@ -106,7 +106,17 @@ function MendelMiddleware(opts) {
         var pack = bpack({raw: true, hasExports: true});
         var decodedResults = trees.findTreeForHash(params.bundle, params.hash);
         if (!decodedResults || decodedResults.error) {
-            return notFound(res, decodedResults && decodedResults.error);
+            return notFound(res, decodedResults && decodedResults.error, params);
+        }
+
+        var modules = indexedDeps(decodedResults.deps.filter(Boolean));
+
+        if (!modules.length) {
+            // Something wrong, modules shouldn't be zero
+            return notFound(res, {
+                code: 'EMPTYBUNDLE',
+                message: 'Tree contents are empty'
+            }, params);
```

Comment:
I agree that this change is not fixing the issue. It will help us to find if this scenario is causing empty bundle issue. Since there are no Mendel logs in production and no way to debug this case on a pod, we need this change to catch this use case in production logs. Also, this will prevent sending empty bundle with 200 status code.

We have checked all available logs in when empty bundle was generated. We couldn't find related error messages or call stacks from mendel. It could be a bad manifest which is not uploaded properly to production. This change will prevent that scenario too.

As you mentioned, the actual issue still exists in our code. I want to prevent this issue and work on a solution. I don't want to keep the issue on production while I am working on fix.